### PR TITLE
Fixing channel issue with ELRS Mavlink-rc: number of channels is not set by default

### PR DESCRIPTION
**Pull request by msoos**
## Description 

If you are using ELRS MAVLINK setup as per: https://www.expresslrs.org/software/mavlink/ then the number of channels will never be set. Hence it will remain 0 -- even though there is clearly control data going through. This is because ELRS is not sending a full `RC_CHANNELS` packet, only channel update packets via `RC_CHANNELS_RAW`, to update the position of the values. Therefore, the number of channels is not set.

This ELRS system forwards BOTH the RC and the Mavlink on the same ELRS link. Using one TX and one RX, you get to have both Mavlink and RC. The Mavlink is forwarded via the backpack chip in the TX to the wifi. You can then connect to this wifi and QGC picks it up automatically, via zero-conf (I think! Very cool) and "just works". Which is great. But the RC doesn't SEEM to work, because the number of channels is never set.

TL;DR: `RC_CHANNELS` is never sent, only `RC_CHANNELS_RAW`, so `chancount` is never set.

## Test Steps

* Using RX ELRS 3.5.0 (release version, not experimental)
* Using TX ELRS 3.5.0 (release version, not experimental)
* Using TX ELRS Backpack 1.5.0 (release version, not experimental)
* Using SpeedyBee F405 Wing (full, not mini) board, Serial 1 connected to ELRS receiver, set up as per: https://www.expresslrs.org/software/mavlink/

We get connectivity of Mavlink forwarded via ELRS + backpack in the TX via WiFi, however, the radio channels cannot be calibrated, and it cannot arm, because the number of channels is seen as 0. With this fix, calibration works and the craft can be armed.

## How to Reproduce
Connect as per above. Observe that calibration is not possible, the QGC says there is no controller connected.

Checklist:
----------
- [x] [Review Contribution Guidelines](https://github.com/mavlink/qgroundcontrol/blob/master/.github/CONTRIBUTING.md).
- [x] [Review Code of Conduct](https://github.com/mavlink/qgroundcontrol/blob/master/.github/CODE_OF_CONDUCT.md).
- [x] I have tested my changes.

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by DonLakeFlyer** [comment]
This seems like a bad hack which goes against mavlink spec. Why can't whatever this system is be fixed to respect the mavlink spec?

https://github.com/mavlink/qgroundcontrol/blob/master/src/FirmwarePlugin/PX4/PX4ParameterMetaData.cc#L78. 

RC_CHANNEL.channel_count - Total number of RC channels being received. This can be larger than 18, indicating that more channels are available but not given in this message. This value should be 0 when no RC channels are available.

**Comment by msoos** [comment]
Thanks for the response @DonLakeFlyer ! It turns out, they send `RC_CHANNELS_OVERRIDE` messages only (not _RAW). Does that change your mind regarding what should be done here?

**Comment by DonLakeFlyer** [comment]
Hmm, that certainly is a mess. 

I could live with this:
* If RC_CHANNEL.channel_count == 0 then look at the individual chan*_raw values. Count up the number of ones which are not UINT16_MAX. That should be the channel count.
* Comment all this as being a hack to support ELRS (refer to issue below in the comment)
* Create an Issue about all of this with a note to remove it once ELRS is fixed to be correct to mavlink spec

That would work?

**Comment by DonLakeFlyer** [comment]
Any updates on this?

**Comment by msoos** [comment]
@DonLakeFlyer Yes, let me do this. Sorry for the delay.

**Comment by msoos** [comment]
Hi @DonLakeFlyer ,

Can you please check the PR now? It now has your requirements implemented. I checked, it works well. Please merge if you think it's good :)

Thanks,

Mate

File: src/Vehicle/Vehicle.cc [hunk 1]
```diff
@@ -1452,6 +1452,17 @@ void Vehicle::_handleRCChannels(mavlink_message_t& message)
     };
     int pwmValues[QGCMAVLink::maxRcChannels];
 
+    // Below is a hack that's needed by ELRS
+    // ELRS is not sending a full RC_CHANNELS packet, only channel update
+    // packets via RC_CHANNELS_RAW, to update the position of the values.
+    // Therefore, the number of channels is not set.
+    if (channels.chancount == 0) {
+        uint8_t num_non_uint16_max = 0;
```
Comment (DonLakeFlyer) [inline src/Vehicle/Vehicle.cc:1460]: Can you remove this. It isn't used.

**Review by DonLakeFlyer — CHANGES_REQUESTED** [review]

File: src/Vehicle/Vehicle.cc (hunk 1 above)
Comment (msoos) [reply src/Vehicle/Vehicle.cc]: OOps, yes, fixed! Sorry.

**Comment by DonLakeFlyer** [comment]
All looks good now. Can you squash down to a single commit. Then once the rebuild finishes I'll merge.

**Comment by msoos** [comment]
Nice, thank you so much! I just squashed :) Let me know how it goes :) Thanks so much,

Mate

**Comment by DonLakeFlyer** [comment]
@msoos Thanks for fixing this

**Comment by msoos** [comment]
Thanks so much for merging!
